homeassistant/components/zha/binary_sensor.py

```diff
@@ -94,51 +85,39 @@
     """Set up the ZHA binary sensors."""
     entities = []
     for discovery_info in discovery_infos:
-        entities.append(BinarySensor(**discovery_info))
+        zha_dev = discovery_info["zha_device"]
+        channels = discovery_info["channels"]
+
+        entity = ZHA_ENTITIES.get_entity(DOMAIN, zha_dev, channels, BinarySensor)
+        if entity:
+            entities.append(entity(**discovery_info))
 
     async_add_entities(entities, update_before_add=True)
 
 
+@STRICT_MATCH(MatchRule(channel_names={CHANNEL_ACCELEROMETER}))
+@STRICT_MATCH(MatchRule(channel_names={CHANNEL_OCCUPANCY}))
+@STRICT_MATCH(MatchRule(channel_names={CHANNEL_ON_OFF}))
 class BinarySensor(ZhaEntity, BinarySensorDevice):
```

Comment:
See the name comment below

Reply:
addressed in https://github.com/home-assistant/home-assistant/pull/30138/commits/e7a145328ca67d684d31598148a1ed9a9ea98fb6